llvm/lib/Analysis/Loads.cpp

```diff
@@ -168,7 +170,7 @@ static bool isDereferenceableAndAlignedPointer(
                                               Size, DL, CtxI, AC, DT, TLI,
                                               Visited, MaxDepth);
 
-  if (CtxI) {
+  if (CtxI && (!UseDerefAtPointSemantics || !V->canBeFreed())) {
```

Comment:
Can we make this unconditional, regardless of the value of UseDerefAtPointSemantics? The way I'd view this is that UseDerefAtPointSemantics is for dereferenceable argument attributes. We can/should make assumptions always use at-point semantics.

Reply:
My plan was first making things correct with the flag, then change the behavior for the attribute in assumes + langref change separately.

I can also pull it in here, although I still need to inspect the other uses of dereferenceable assumptions.

Reply:
Okay, that's fin by me.